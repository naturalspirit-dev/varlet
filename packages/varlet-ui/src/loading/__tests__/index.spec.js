const Loading = require('../../../cjs/loading').default
const { render } = require('@testing-library/vue')

test('test loading', async () => {
	const wrapper = render(Loading)
	console.log(wrapper)
})
