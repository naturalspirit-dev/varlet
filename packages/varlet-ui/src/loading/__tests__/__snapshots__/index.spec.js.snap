// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test circle props 1`] = `
"<div class=\\"var--box var-loading\\" radius=\\"12\\">
  <div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block var-loading__circle-block-normal\\" style=\\"color: rgb(41, 121, 255);\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`test cube props 1`] = `
"<div class=\\"var--box var-loading\\">
  <!--v-if-->
  <!--v-if-->
  <div class=\\"var-loading__cube var-loading__cube-small\\">
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__cube-item var-loading__cube-item-small\\"></div>
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__cube-item var-loading__cube-item-small\\"></div>
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__cube-item var-loading__cube-item-small\\"></div>
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__cube-item var-loading__cube-item-small\\"></div>
  </div>
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`test disappear props 1`] = `
"<div class=\\"var--box var-loading\\">
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  <div class=\\"var-loading__disappear var-loading__disappear-small\\">
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__disappear-item var-loading__disappear-item-small\\"></div>
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__disappear-item var-loading__disappear-item-small\\"></div>
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__disappear-item var-loading__disappear-item-small\\"></div>
  </div>
</div>"
`;

exports[`test loading example 1`] = `
"<div class=\\"loading-demo\\">
  <div class=\\"loading-demo__type\\">
    <div class=\\"app-type\\">类型</div>
    <div class=\\"loading-demo__type-block\\">
      <div class=\\"var--box var-loading\\">
        <div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block var-loading__circle-block-normal\\" style=\\"color: currentColor;\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      <div class=\\"var--box var-loading\\">
        <!--v-if-->
        <!--v-if-->
        <div class=\\"var-loading__cube var-loading__cube-normal\\">
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item-normal\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item-normal\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item-normal\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item-normal\\"></div>
        </div>
        <!--v-if-->
        <!--v-if-->
      </div>
      <div class=\\"var--box var-loading\\">
        <!--v-if-->
        <div class=\\"var-loading__wave var-loading__wave-normal\\">
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item-normal\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item-normal\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item-normal\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item-normal\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item-normal\\"></div>
        </div>
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      <div class=\\"var--box var-loading\\">
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <div class=\\"var-loading__rect var-loading__rect-normal\\">
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div>
        </div>
        <!--v-if-->
      </div>
      <div class=\\"var--box var-loading\\">
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <div class=\\"var-loading__disappear var-loading__disappear-normal\\">
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\"></div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"loading-demo__type\\">
    <div class=\\"app-type\\">颜色</div>
    <div class=\\"loading-demo__type-block\\">
      <div class=\\"var--box var-loading\\">
        <div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block var-loading__circle-block-normal\\" style=\\"color: rgb(58, 122, 254);\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      <div class=\\"var--box var-loading\\">
        <!--v-if-->
        <!--v-if-->
        <div class=\\"var-loading__cube var-loading__cube-normal\\">
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__cube-item var-loading__cube-item-normal\\"></div>
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__cube-item var-loading__cube-item-normal\\"></div>
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__cube-item var-loading__cube-item-normal\\"></div>
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__cube-item var-loading__cube-item-normal\\"></div>
        </div>
        <!--v-if-->
        <!--v-if-->
      </div>
      <div class=\\"var--box var-loading\\">
        <!--v-if-->
        <div class=\\"var-loading__wave var-loading__wave-normal\\">
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__wave-item var-loading__wave-item-normal\\"></div>
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__wave-item var-loading__wave-item-normal\\"></div>
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__wave-item var-loading__wave-item-normal\\"></div>
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__wave-item var-loading__wave-item-normal\\"></div>
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__wave-item var-loading__wave-item-normal\\"></div>
        </div>
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      <div class=\\"var--box var-loading\\">
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <div class=\\"var-loading__rect var-loading__rect-normal\\">
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div>
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div>
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div>
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div>
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div>
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div>
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div>
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__rect-item var-loading__rect-item-normal\\"></div>
        </div>
        <!--v-if-->
      </div>
      <div class=\\"var--box var-loading\\">
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <div class=\\"var-loading__disappear var-loading__disappear-normal\\">
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\"></div>
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\"></div>
          <div style=\\"background-color: rgb(58, 122, 254);\\" class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\"></div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"loading-demo__size\\">
    <div class=\\"app-type\\">大小</div>
    <div class=\\"loading-demo__size-block\\">
      <div class=\\"var--box var-loading\\">
        <div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block var-loading__circle-block-small\\" style=\\"color: currentColor;\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      <div class=\\"var--box var-loading\\">
        <!--v-if-->
        <!--v-if-->
        <div class=\\"var-loading__cube var-loading__cube-small\\">
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item-small\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item-small\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item-small\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__cube-item var-loading__cube-item-small\\"></div>
        </div>
        <!--v-if-->
        <!--v-if-->
      </div>
      <div class=\\"var--box var-loading\\">
        <!--v-if-->
        <div class=\\"var-loading__wave var-loading__wave-small\\">
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item-small\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item-small\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item-small\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item-small\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__wave-item var-loading__wave-item-small\\"></div>
        </div>
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      <div class=\\"var--box var-loading\\">
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <div class=\\"var-loading__rect var-loading__rect-small\\">
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-small\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-small\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-small\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-small\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-small\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-small\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-small\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__rect-item var-loading__rect-item-small\\"></div>
        </div>
        <!--v-if-->
      </div>
      <div class=\\"var--box var-loading\\">
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        <div class=\\"var-loading__disappear var-loading__disappear-small\\">
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__disappear-item var-loading__disappear-item-small\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__disappear-item var-loading__disappear-item-small\\"></div>
          <div style=\\"background-color: currentColor;\\" class=\\"var-loading__disappear-item var-loading__disappear-item-small\\"></div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`test rect props 1`] = `
"<div class=\\"var--box var-loading\\">
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  <div class=\\"var-loading__rect var-loading__rect-small\\">
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item-small\\"></div>
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item-small\\"></div>
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item-small\\"></div>
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item-small\\"></div>
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item-small\\"></div>
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item-small\\"></div>
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item-small\\"></div>
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__rect-item var-loading__rect-item-small\\"></div>
  </div>
  <!--v-if-->
</div>"
`;

exports[`test wave props 1`] = `
"<div class=\\"var--box var-loading\\">
  <!--v-if-->
  <div class=\\"var-loading__wave var-loading__wave-small\\">
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__wave-item var-loading__wave-item-small\\"></div>
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__wave-item var-loading__wave-item-small\\"></div>
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__wave-item var-loading__wave-item-small\\"></div>
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__wave-item var-loading__wave-item-small\\"></div>
    <div style=\\"background-color: rgb(41, 121, 255);\\" class=\\"var-loading__wave-item var-loading__wave-item-small\\"></div>
  </div>
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</div>"
`;
